src/Rector/v12/v0/typo3/RemoveRedundantFeLoginModeMethodsRector.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ssch\TYPO3Rector\Rector\v12\v0\typo3;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Type\ObjectType;
+use Rector\Core\Console\Output\RectorOutputStyle;
+use Rector\Core\Rector\AbstractRector;
+use Ssch\TYPO3Rector\Helper\Typo3NodeResolver;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.0/Breaking-96616-RemoveFrontendLoginModeForPages.html
+ * @see \Ssch\TYPO3Rector\Tests\Rector\v12\v0\typo3\RemoveRedundantFeLoginModeMethodsRector\RemoveRedundantFeLoginModeMethodsRectorTest
+ */
+final class RemoveRedundantFeLoginModeMethodsRector extends AbstractRector
+{
+    public function __construct(
+        private readonly RectorOutputStyle $rectorOutputStyle,
+        private readonly Typo3NodeResolver $typo3NodeResolver
+    ) {
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    /**
+     * @param Node\Expr\MethodCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->shouldSkip($node)) {
+            return null;
+        }
+
+        if ($this->isName($node->name, 'checkIfLoginAllowedInBranch')) {
+            $this->rectorOutputStyle->note(
+                'Please remove the usage of TypoScriptFrontendController->checkIfLoginAllowedInBranch(). Also check the changelog file `Breaking-96616-RemoveFrontendLoginModeForPages` for further migration advice.'
```

Comment:
A link would be much better here than just mentioning the file where I would have no glue where to look for this file

Reply:
I would prefer that as well but we cannot ensure those links will work as they restructure the docs often.
With the raw file name you can check within your local projects or simply google to find the fitting link.

Also: it is mentioned that the use case for goes close to non, so i would think that this rule only is applied in absolute edge cases..